shared/fs/filepreview/text-view.js.flow

```diff
@@ -3,6 +3,8 @@ import * as React from 'react'
 
 export type TextViewProps = {
   url: string,
+  onInvalidToken: () => void,
 }
 
-declare export default React.ComponentType<TextViewProps>
+// Had to use "any" here to make flow work. Weird!
+declare export default React.ComponentType<any>
```

Comment:
This was super weird. `image-view.js.flow` has a similar thing and is used similarly, but flow doesn't complain even if it stays that way. Could somebody help me understand why having this typed works for `ImageView` but not `TextView`?

Here's the flow error if the props here were `TextViewProps` rather than `any`:

```
yarn run v1.3.2
$ /Users/songgao/gopath/src/github.com/keybase/client/shared/node_modules/.bin/flow
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ fs/filepreview/default-view-container.js:10:46

Property path is missing in TextViewProps [1].

     fs/filepreview/default-view-container.js
      7│ import DefaultView from './default-view'
      8│ import {navigateAppend} from '../../actions/route-tree'
      9│
     10│ const mapStateToProps = (state: TypedState, {path}) => {
     11│   const pathItem = state.fs.pathItems.get(path) || Constants.makeUnknownPathItem()
     12│   const _username = state.config.username || undefined
     13│   return {

     flow-typed/npm/react-redux_v5.x.x.js
 [1] 90│   ): (component: Com) => ComponentType<$Diff<ElementConfig<Com>, RMP> & SP & DP & MP>;


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ fs/filepreview/view-container.js:33:55

Property path is missing in TextViewProps [1].

     fs/filepreview/view-container.js
     30│   onInvalidToken: () => dispatch(FsGen.createRefreshLocalHTTPServerInfo()),
     31│ })
     32│
     33│ const mergeProps = ({_serverInfo}, {onInvalidToken}, {path}) => ({
     34│   url: Constants.generateURL(path, _serverInfo.address, _serverInfo.token),
     35│   onInvalidToken,
     36│ })

     flow-typed/npm/react-redux_v5.x.x.js
 [1] 90│   ): (component: Com) => ComponentType<$Diff<ElementConfig<Com>, RMP> & SP & DP & MP>;


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ fs/filepreview/view-container.js:42:12

Property ft is missing in TextViewProps [1].

     fs/filepreview/view-container.js
     39│
     40│ export default compose(
     41│   mapProps(({path, fileViewType}: Props) => ({path, ft: fileViewType || Constants.viewTypeFromPath(path)})),
     42│   branch(({ft}) => ft === 'default', renderComponent(DefaultView)),
     43│   branch(({ft}) => ft === 'text', renderComponent(httpConnect(TextView))),
     44│   branch(({ft}) => ft === 'image', renderComponent(httpConnect(ImageView)))
     45│ )(() => <Text type="BodyError">This shouldn't happen</Text>)

     flow-typed/npm/react-redux_v5.x.x.js
 [1] 90│   ): (component: Com) => ComponentType<$Diff<ElementConfig<Com>, RMP> & SP & DP & MP>;


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ fs/filepreview/view-container.js:43:12

Property ft is missing in TextViewProps [1].

     fs/filepreview/view-container.js
     40│ export default compose(
     41│   mapProps(({path, fileViewType}: Props) => ({path, ft: fileViewType || Constants.viewTypeFromPath(path)})),
     42│   branch(({ft}) => ft === 'default', renderComponent(DefaultView)),
     43│   branch(({ft}) => ft === 'text', renderComponent(httpConnect(TextView))),
     44│   branch(({ft}) => ft === 'image', renderComponent(httpConnect(ImageView)))
     45│ )(() => <Text type="BodyError">This shouldn't happen</Text>)
     46│

     flow-typed/npm/react-redux_v5.x.x.js
 [1] 90│   ): (component: Com) => ComponentType<$Diff<ElementConfig<Com>, RMP> & SP & DP & MP>;


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ fs/filepreview/view-container.js:44:12

Property ft is missing in TextViewProps [1].

     fs/filepreview/view-container.js
     41│   mapProps(({path, fileViewType}: Props) => ({path, ft: fileViewType || Constants.viewTypeFromPath(path)})),
     42│   branch(({ft}) => ft === 'default', renderComponent(DefaultView)),
     43│   branch(({ft}) => ft === 'text', renderComponent(httpConnect(TextView))),
     44│   branch(({ft}) => ft === 'image', renderComponent(httpConnect(ImageView)))
     45│ )(() => <Text type="BodyError">This shouldn't happen</Text>)
     46│

     flow-typed/npm/react-redux_v5.x.x.js
 [1] 90│   ): (component: Com) => ComponentType<$Diff<ElementConfig<Com>, RMP> & SP & DP & MP>;



Found 5 errors
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
don't you want
```
declare export default class ImageView extends React.Component<ImageViewProps> {}
and
declare export default class TextView extends React.Component<TextViewProps> {}

```

Reply:
Yeah that's what I tried initially, but got above flow error.

Reply:
It seems to be caused by `branch`. After replacing that flow is happy here.